### PR TITLE
Ruff (no formatter)

### PR DESCRIPTION
**Pull request by Cadair**
Use ruff over flake8.

This swaps out flake8 for ruff.

### CODE REVIEW

File: .ruff.toml
Comment (Cadair) [reply .ruff.toml:8]: That would be nice.

**Review by SolarDrew — APPROVED** [review]
Seems fine :+1:  I look forward to resolving conflicts for the rest of the year.

**Comment by Cadair** [comment]
Let's hold off merging this until #344 lands.

**Comment by Cadair** [comment]
pre-commit.ci autofix

**Pull request merged**
